include/rnp/rnp_sdk.h

```diff
@@ -41,13 +41,21 @@ extern "C" {
 
 #include <rnp/rnp_def.h>
 
-#ifndef PRINTFLIKE
-#define PRINTFLIKE(n, m) __attribute__((format(printf, n, m)))
-#endif
-
 typedef enum { RNP_HEX_LOWERCASE, RNP_HEX_UPPERCASE } rnp_hex_format_t;
 
+#if _MSC_VER >= 1400
+#include <sal.h>
+#if _MSC_VER > 1400
+void rnp_log(_Printf_format_string_ const char *, ...);
+#else
+void rnp_log(__format_string const char *, ...);
+#endif /* FORMAT_STRING */
+#else  /* _MSC_VER */
+#ifndef PRINTFLIKE
+#define PRINTFLIKE(n, m) __attribute__((format(printf, n, m)))
+#endif /* PRINTFLIKE */
 void rnp_log(const char *, ...) PRINTFLIKE(1, 2);
```

Comment:
We do not use this `rnp_log()` anywhere, it seems to be an artifact of the ancient times. So it, and PRINTFLIKE could be safely deleted.